test/percy-agent-client/dom.test.ts

```diff
@@ -0,0 +1,260 @@
+        const exampleDOM = createExample('<div class="box"></div>')
+
+        createCSSOM()
+        dom = new DOM(exampleDOM, { enableJavaScript: true })
+      })
+
+      it('does not serialize the CSSOM when JS is enabled', () => {
+        const cssomOwnerNode = document.styleSheets[0].ownerNode as any
+
+        expect(cssomOwnerNode.innerText).to.equal('')
+        expect(dom.clonedDOM.querySelectorAll('[data-percy-cssom-serialized]').length).to.equal(0)
+      })
+    })
+
+    describe('inputs', () => {
+      let $domString: any
+
+      beforeEach(async () => {
+        const example = createExample(`
+          <form>
+            <label for="name">Name</label>
+            <input id="name" type="text" />
+
+            <input id="mailing" type="checkbox" />
+            <label for="mailing">Subscribe?</label>
+
+            <input id="radio" type="radio" />
+            <label for="radio">Radio</label>
+
+            <input id="nevercheckedradio" type="radio" />
+            <label for="nevercheckedradio">Never checked</label>
+
+            <label for="feedback">Feedback</label>
+            <textarea id="feedback"></textarea>
+          </form>
+        `)
+
+        await type('#name', 'Bob Boberson')
+        await type('#feedback', 'This is my feedback... And it is not very helpful')
+        await check('#radio')
+        await check('#mailing')
+
+        dom = new DOM(example)
+        $domString = cheerio.load(dom.snapshotString())
+      })
+
+      it('serializes checked checkboxes', () => {
+        expect($domString('#mailing').attr('checked')).to.equal('checked')
+      })
+
+      it('serializes checked radio buttons', () => {
+        expect($domString('#radio').attr('checked')).to.equal('checked')
+      })
+
+      it('serializes textareas', () => {
+        expect($domString('#feedback').attr('value')).to.equal(
+          'This is my feedback... And it is not very helpful',
+        )
+      })
+
+      it('serializes input elements', () => {
+        expect($domString('#name').attr('value')).to.equal('Bob Boberson')
+      })
+
+      it('adds a guid data-attribute to the original DOM', () => {
+        expect(document.querySelectorAll('[data-percy-element-id]').length).to.equal(5)
+      })
+
+      it('adds matching guids to the orignal DOM and cloned DOM', () => {
+        const originalElementIds = document.querySelectorAll('[data-percy-element-id]')
+        const cloneElementIds = dom.clonedDOM.querySelectorAll('[data-percy-element-id]')
+        const originalValue = originalElementIds[0].attributes['data-percy-element-id'].value
+        const cloneValue = cloneElementIds[0].attributes['data-percy-element-id'].value
+
+        expect(originalValue).to.equal(cloneValue)
+      })
+
+      it('does not mutate values in origial DOM', () => {
+        const originalDOMInput: any = document.querySelector('#name')
+
+        expect(originalDOMInput.attributes.value).to.equal(undefined)
```

Comment:
```suggestion
        expect(originalDOMInput.getAttribute('value')).to.equal(undefined)
```

I think your intention is something like this?